wdl-analysis/src/types/v1.rs

```diff
@@ -1644,10 +1670,11 @@ impl<'a, C: EvaluationContext> ExprTypeEvaluator<'a, C> {
     ) {
         let (label, span, fix) = match target.text() {
             "select_first" => {
-                let ty = &arguments[0]
+                let ty = arguments[0]
```

Comment:
Suggested fix is to just `if let Some(ty) = arguments[0].as_array().map(|a| a.element_type()) { ... }`

Reply:
For the check for `defined` below, we'll also want to not figure the diagnostic if the argument is also `Union`, so ` arguments[0].is_optional() || arguments[0].is_union()`.